apps/etl-func/src/adapters/functions/messages-ingestion.ts

```diff
@@ -11,16 +11,22 @@ const logger = pino({
   level: process.env.NODE_ENV === "production" ? "error" : "debug",
 });
 
+const processMessageMetadata = (
+  input: unknown,
+): MessageMetadata | undefined => {
+  const { data, success } = messageMetadataSchema.safeParse(input);
+  return success && !data?.isPending ? data : undefined;
+};
+
 const messagesIngestionHandler =
   (ingestUseCase: IngestMessageUseCase): CosmosDBHandler =>
   async (documents: unknown[]) => {
     //Avoiding all documents different from MessageMetadata schema and with
     //isPending equals to true
-    const messagesMetadata: MessageMetadata[] = documents.filter(
-      (item): item is MessageMetadata => {
-        const parsedItem = messageMetadataSchema.safeParse(item);
-        return parsedItem.success && !parsedItem.data.isPending;
-      },
+
+    const parsedMessagesMetadata = documents.map(processMessageMetadata);
+    const messagesMetadata: MessageMetadata[] = parsedMessagesMetadata.filter(
+      (item): item is MessageMetadata => item !== undefined,
```

Comment:
Can't we unify the code like this by removing the function and linking the map and the filter together?
```suggestion
const messagesMetadata: MessageMetadata[] = documents
  .map((input) => {
    const { data, success } = messageMetadataSchema.safeParse(input);
    return success && !data?.isPending ? data : undefined;
  })
  .filter((item): item is MessageMetadata => item !== undefined);
```

Reply:
I think in this way we are going to loose readability. Do you see any problem if we keep the current version ?